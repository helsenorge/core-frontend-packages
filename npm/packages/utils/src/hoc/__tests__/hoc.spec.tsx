import { render, screen } from '@testing-library/react';
import { vi } from 'vitest';

import mountHOC from '../mount';
import RenderToBody from '../render-to-body';

describe('HOC utils', () => {
  describe('Gitt at en komponent wrappes i mount HOC', () => {
    describe('Når den instansieres', () => {
      it('Så mounter den komponenten og kaller mount prop', () => {
        const MountExample = mountHOC(() => <div>{'test'}</div>);
        const mountMock = vi.fn();

        render(<MountExample mount={mountMock} />);

        const text = screen.getByText('test');
        expect(text).toBeVisible();

        expect(mountMock).toHaveBeenCalled();
      });
    });
  });

  describe('Gitt at en komponent wrappes i render-to-body HOC', () => {
    describe('Når printable er false', () => {
      it('Så rendres det en portal', () => {
        render(
          <RenderToBody>
            <section>{'My first section rendered to document'}</section>
          </RenderToBody>
        );

        expect(document.body).toMatchSnapshot();
      });
    });

    describe('Når printable er true', () => {
      it('Så rendres det en portal med body tag', () => {
        render(
          <RenderToBody printable>
            <section>{'My second section rendered to body (printable)'}</section>
          </RenderToBody>
        );

        expect(document.body).toMatchSnapshot();
      });
    });
  });
});
