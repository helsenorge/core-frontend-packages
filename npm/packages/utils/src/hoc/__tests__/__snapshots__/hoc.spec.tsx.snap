// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC utils Gitt at en komponent wrappes i render-to-body HOC Når printable er false Så renderes det en portal 1`] = `
<RenderToBody>
  <Portal
    containerInfo={
      <div>
        <section>
          My first section rendered to document
        </section>
      </div>
    }
  >
    <section>
      My first section rendered to document
    </section>
  </Portal>
</RenderToBody>
`;

exports[`HOC utils Gitt at en komponent wrappes i render-to-body HOC Når printable er true Så renderes det en portal med body tag 1`] = `
<RenderToBody
  printable={true}
>
  <Portal
    containerInfo={
      <body>
        <div>
          <section>
            My first section rendered to document
          </section>
        </div>
        <section>
          My second section rendered to body (printable)
        </section>
      </body>
    }
  >
    <section>
      My second section rendered to body (printable)
    </section>
  </Portal>
</RenderToBody>
`;

exports[`HOC utils Gitt at en komponent wrappes i track-route-change HOC Når den instansieres Så kaller den adobe-analytics og rendres riktig 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(TrackRouteChange)>
      <TrackRouteChange
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <section>
          Track routing section
        </section>
      </TrackRouteChange>
    </withRouter(TrackRouteChange)>
  </Router>
</BrowserRouter>
`;
